### ISSUE TITLE
Add VELO material model to RapidSim and ability to use it to compute Brems

### ISSUE THREAD

**Issue by particleist** [gcowan/RapidSim#30]
We should incorporate the VELO material model from Mike&Phil as an option. Once this is done we can have Brems which properly correlates to a track's position in the VELO, and we can then smear and save the Brems photons which the material model creates as it goes along. This will also open up the ability to smear tracks according to their multiple scattering in the VELO, and thus have a proper smearing of SVs. I'll make a new branch and dump the code I got from Phil onto it, and we can then start discussing how much refactoring it will need, since the coding style and conventions have nothing to do with the ones used in RapidSim...

**Comment by bifani** [comment]
Awesome!

I remember you presented some preliminary result at an R(X) meeting that showed this method systematically generates too little brem compared to what is in the full simulation
Is this still the case?

You also mentioned two possible reasons:
1) missing material (TT/RICH1)
2) mismodelling the brem for a given material

For 1), the Birmingham group developed an MC test where one can study the amount of material traversed by a particle between two planes placed in the detector as a function of e.g. (x,y)
I still need to understand how flexible and up-to-date the tool is, but do you think this could be a useful, although simplified, parameterisation of the material after the velo?

**Comment by particleist** [comment]
Yes this still undergenerates Brems. I think the most important thing for a RICH/TT/UT model is that the format of how this is described is the same one which Phil uses for the VELO, so that it can be plugged in as "just another layer" into the code. Then how the model is generated into this data format is another story, if you have a tool which can be used for this fantastic. I would suggest you liaise with Phil and also Yasmine's student Vitalii who is looking at this I think.

**Comment by bifani** [comment]
ok, I don't think that this tool can produce a model as detailed as Phil's tool but I will check with Nigel and the students to better understand what can be done

**Comment by particleist** [comment]
I don't think it needs to be as detailed necessarily, we can approximate the TT/UT and RICH as single blocks of a certain average thickness I would have thought?

**Comment by gcowan** [comment]
This is what I was thinking. Should be good enough.

**Comment by bifani** [comment]
sorry, I probably misinterpreted what you meant by "same format as the VELO"

what I think can be done is to produce an (x,y) map containing the average material traversed by a particle between two z coordinates
depending how easy/feasible it is to move the sampling planes it might also be possible to e.g. parameterise independently TT and RICH

**Comment by particleist** [comment]
"same format" means the same root file format as the root files from Phil describing the VELO geometry, nothing more. I don't actually know what that is. :)

**Comment by vlisovsk** [comment]
In the question of brem emission in different subdetectors, we should probably refer to the map of BremOrigin coordinates in the full simulation. 
E.g. the one from pKee MC:
/afs/cern.ch/work/v/vlisovsk/public/BremOrigin.pdf

**Comment by bifani** [comment]
Hi,
we managed to resurrect the RadLenght MC test, however, there are still a couple of things that need to be checked/understood

Anyways, the plots below show the kind of information that this tool can provide, in particular, the average number of x0 traversed by the tracks between the end of Velo and the beginning of the Magnet (effectively corresponding to RICH1+TT)

I don't know the details of how the Velo is dealt with, but do you think this type of histo would be useful to model the remaining brem before the magnet?

PS. the empty bins are just due to low stats

![rich1_tt_1d](https://user-images.githubusercontent.com/17066912/33833263-695905aa-de7f-11e7-8318-15f27fae974a.png)
![rich1_tt_2d](https://user-images.githubusercontent.com/17066912/33833264-697110fa-de7f-11e7-9b63-aafc2426ba17.png)

**Comment by particleist** [comment]
I think that's exactly the information we need, since the classes from Phil/Mike essentially take the material thickness & type as a fundamental input. The propagation of the track through this material is universal whether VELO or what have you.

**Comment by bifani** [comment]
OK good, in case you or @vlisovsk want to try using this in RapidSim I can provide a root file with the above plots
Meanwhile, we will make sure that we are accumulating the radiation length over the correct z region (sampling planes cannot always be added where we'd want to due to GEANT4 not liking overlapping volumes)

**Comment by vlisovsk** [comment]
So I guess this requires the modification of ProcessBrem function, and the way how the cross-section is computed.
In the current code, I see the Material class computing the `fc` quantity which is formula 33.27 in the PDG2016 together with `Lrad` and `Lrad'` (formulae in the table 33.2); and then the differential cross-section `ProcessBrem::dsigma` which is (rewritten in another way) formula 33.29.
So I see the cross-section can be easily rewritten in terms of the radiation length taken from Simone's map, however there is a remaining factor containing A and one term containing Z of the material. 
To get rid of the Z-term, one may use the approximation by formula 33.30 already expressed in terms of radiation length, however, A still remains there.
Can we just say that average A of the material of detector is some 27 (AFAIR Si=28, Al=26)? 
Then it should be easy to use this map together with formula 33.30 where `X0` is taken from the map, A is 27, `k` is called `eg` in the code, and `y` is `eg/ec`.
If we get something close to the proper description, we may tune the average A as a free parameter.
What do you think?
